benchmarks/python/utils.py

```diff
@@ -0,0 +1,130 @@
+import argparse
+import json
+import math
+import random
+from enum import Enum
+from statistics import mean
+from typing import List
+
+import numpy as np
+
+
+class ChosenAction(Enum):
+    GET_NON_EXISTING = 1
+    GET_EXISTING = 2
+    SET = 3
+
+
+PORT = 6379
+PROB_GET = 0.8
+PROB_GET_EXISTING_KEY = 0.8
+SIZE_GET_KEYSPACE = 3750000  # 3.75 million
+SIZE_SET_KEYSPACE = 3000000  # 3 million
+
+
+def create_argument_parser():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--resultsFile",
+        help="Where to write the results file",
+        required=False,
+        default="../results/python-results.json",
+    )
+    parser.add_argument(
+        "--dataSize", help="Size of data to set", required=False, default="100"
+    )
+    parser.add_argument(
+        "--clients", help="Which clients should run", required=False, default="all"
+    )
+    parser.add_argument(
+        "--host", help="What host to target", required=False, default="localhost"
+    )
+    parser.add_argument(
+        "--clientCount",
+        help="Number of clients to run concurrently",
+        nargs="+",
+        required=False,
+        default=("1"),
+    )
+    parser.add_argument(
+        "--tls",
+        help="Should benchmark a TLS server",
+        action="store_true",
+        required=False,
+        default=False,
+    )
+    parser.add_argument(
+        "--clusterModeEnabled",
+        help="Should benchmark a cluster mode enabled cluster",
+        action="store_true",
+        required=False,
+        default=False,
+    )
+    parser.add_argument(
+        "--port",
+        default=PORT,
+        type=int,
+        required=False,
+        help="Which port to connect to, defaults to `%(default)s`",
+    )
+    parser.add_argument(
+        "--minimal", help="Should run a minimal benchmark", action="store_true"
+    )
+    parser.add_argument(
+        "--concurrentTasks",
+        help="List of number of concurrent tasks/threads to run",
```

Comment:
please write that tasks for async and threads for sync (bc it can be confusing - the async bench can't be run with multiple threads)